### PR TITLE
meta tests: refactor run_pytest

### PR DESCRIPTION
**Pull request by ikonst**
Factor `run_pytest` out of mypy/test/meta/test_*.py.

Also, prefixing the nested pytest output so it's harder to mix it up with parent pytest's:
```
>> .../mypy/venv/bin/python -m pytest -n 0 -s mypy/test/testcheck.py::TypeCheckSuite::check-meta-909158d4-c9c7-41e6-be50-cf101aaa2b56.test
NESTED PYTEST STDOUT: ============================= test session starts ==============================
                    : platform darwin -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0
                    : rootdir: .../mypy
```

### CODE REVIEW

**Comment by ikonst** [comment]
@hauntsaninja

File: mypy/test/meta/_pytest.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    p_root = p_test_data.parent.parent
+    p = p_test_data / f"{data_file_prefix}-meta-{uuid.uuid4()}.test"
+    assert not p.exists()
+    data_suite = dedent_docstring(data_suite)
+    try:
+        p.write_text(data_suite)
+
+        test_nodeid = f"{pytest_node_prefix}::{p.name}"
+        extra_args = [sys.executable, "-m", "pytest", "-n", "0", "-s", *extra_args, test_nodeid]
+        if sys.version_info >= (3, 8):
```
Comment (hauntsaninja) [inline mypy/test/meta/_pytest.py:49]: We've dropped support for 3.7 :-)

**Review by hauntsaninja — COMMENTED** [review]
Looks good, one nit!

File: mypy/test/meta/_pytest.py (hunk 1 above)
Comment (ikonst) [reply mypy/test/meta/_pytest.py]: b0083b03d1dd12a90e6dc8cae6558f450a3b3527 but we should maybe enable https://beta.ruff.rs/docs/rules/outdated-version-block/ and perhaps more "pyupgrade" rules? (in a separate PR)

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**